ci/jenkins/test.sh

```diff
@@ -839,6 +854,19 @@ function run_conformance {
     kubectl rollout status daemonset/antrea-agent -n kube-system
 
     set +e
+    # Conformance DNS test cases fail in IPv6 Kind cluster because here CoreDNS pods are IPv6 pods 
+    # and it can't reach IPv4 upstream nameserver.
+    # Note: Default CoreDNS ConfigMap uses /etc/resolv.conf to get upstream nameservers listed in resolv.conf file on the host,
+    # the upstream nameservers are accessed by IPv4, when single stack IPv6 pods try to connect IPv4 nameserver it return servfail,
+    # with the new ConfigMap we will forward DNS query to "vmware.com:53" and it will return 'NXDOMAIN' so that client keep searching.
+    # For more detail: https://github.com/kubernetes/kubernetes/issues/94794 
+    if [[ $IP_MODE == "ipv6" && "$TESTCASE" =~ "conformance" && $TESTBED_TYPE == "kind" && $JENKINS_LOCATION == "lab" ]]; then
+        # Override the CoreDNS ConfigMap with new one.
+        kubectl apply -f ./ci/coredns-configmap.yaml
+    elif [[ $IP_MODE == "ipv6" && "$TESTCASE" =~ "conformance" && $TESTBED_TYPE == "kind" && $JENKINS_LOCATION == "cloud" ]]; then  
+        kubectl apply -f ./ci/cloud-coredns-configmap.yaml 
+    fi 
```

Comment:
```suggestion
    if [[ $IP_MODE == "ipv6" && "$TESTCASE" =~ "conformance" && $TESTBED_TYPE == "kind" && $JENKINS_LOCATION == "lab" ]]; then
        # Override the CoreDNS ConfigMap with new one.
        kubectl apply -f ./ci/coredns-configmap.yaml
    elif [[ $IP_MODE == "ipv6" && "$TESTCASE" =~ "conformance" && $TESTBED_TYPE == "kind" && $JENKINS_LOCATION == "cloud" ]]; then  
        kubectl apply -f ./ci/cloud-coredns-configmap.yaml 
    fi 
```
if [[ $IP_MODE == "ipv6" && "$TESTCASE" =~ "conformance" && $TESTBED_TYPE == "kind" ]]; then
    if [[ $JENKINS_LOCATION == "lab" ]]; then
        # Override the CoreDNS ConfigMap with a new one for lab environment.
        kubectl apply -f ./ci/coredns-configmap.yaml
    elif [[ $JENKINS_LOCATION == "cloud" ]]; then
        # Override the CoreDNS ConfigMap with a different one for cloud environment.
        kubectl apply -f ./ci/cloud-coredns-configmap.yaml
    fi
fi